etl/churn.kp/orig_src/Churn.ipynb

```diff
@@ -473,126 +471,206 @@
-    "            # Update the dashboard file\n",
-    "            dash_bucket = \"net-mozaws-prod-metrics-data\"\n",
-    "            dash_s3_name = \"telemetry-churn/{}\".format(churn_outfile)\n",
-    "            transfer.upload_file(churn_outfile, dash_bucket, dash_s3_name,\n",
-    "                                 extra_args={'ACL': 'bucket-owner-full-control'})\n",
-    "        \n",
-    "        print \"Finished week from {} to {} at {}\".format(week_start, week_end, _datetime.utcnow())\n",
-    "    # Move forward by a week\n",
-    "    week_start_date += timedelta(7)\n",
-    "    week_end_date += timedelta(7)\n",
-    "    week_start = fmt(week_start_date)\n",
-    "    week_end = fmt(week_end_date)"
+    "        upload_to_s3(records_df.rdd.collect(), churn_outfile)\n",
+    "\n",
+    "    # Write to s3 as parquet, file size is on the order of 40MB\n",
+    "    parquet_s3_path = (\"s3://{}/{}/v1/week_start={}\"\n",
+    "                       .format(churn_bucket_parquet, churn_s3_path_parquet, week_start))\n",
+    "    print \"{}: Writing output as parquet to {}\".format(_datetime.utcnow(), parquet_s3_path)\n",
+    "    records_df.repartition(1).write.parquet(parquet_s3_path, mode=\"overwrite\")\n",
+    "\n",
+    "    print(\"Finished week from {} to {} at {}\"\n",
+    "          .format(week_start, week_end, _datetime.utcnow()))\n",
+    "\n",
+    "    \n",
+    "def daterange(start_date, end_date):\n",
+    "    for n in range(int((end_date - start_date).days)//7):\n",
+    "        yield (start_date + timedelta(n*7)).strftime(\"%Y%m%d\")\n",
+    "\n",
+    "\n",
+    "def backfill(df, start_date_yyyymmdd):\n",
+    "    \"\"\" Import data from a start date to an end date\"\"\"\n",
+    "    start_date = snap_to_beginning_of_week(\n",
+    "        _datetime.strptime(start_date_yyyymmdd, \"%Y%m%d\"), \n",
+    "        \"Sunday\")\n",
+    "    end_date = _datetime.utcnow() - timedelta(1) # yesterday\n",
+    "    for d in daterange(start_date, end_date):\n",
+    "        try:\n",
+    "            compute_churn_week(df, d)\n",
+    "        except Exception as e:\n",
+    "            print e"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from os import environ\n",
+    "\n",
+    "# Uncomment to set the date manually\n",
+    "# os.environ['date'] = \"20161101\"\n",
+    "\n",
+    "env_date = environ.get('date', None)\n",
+    "if not env_date:\n",
+    "    raise ValueError(\"date not provided\")\n",
+    "\n",
+    "# If this job is scheduled, we need the input date to lag a total of \n",
+    "# 10 days of slack for incoming data + the 7 days used in churn computation, or 17 days\n",
+    "# from the current date. \n",
+    "week_start_date = snap_to_beginning_of_week(\n",
+    "    _datetime.strptime(env_date, \"%Y%m%d\") - timedelta(17),\n",
```

Comment:
How does this behave when date is from an old week? Does it really compute the data for the day we're interested in? AFAIK Airflow could be passing old dates as well, not just the latest one.

Moreover, AFAICT the slack dance is already taken care of by |compute_churn_week|. Is there any other reason to do it here as well?